src/connector/src/source/iceberg/mod.rs

```diff
@@ -662,34 +673,279 @@ pub async fn scan_task_to_chunk(
+            let task_clone: FileScanTask = (*delete_task).clone();
+            let delete_stream = tokio_stream::once(Ok(task_clone));
+            let mut delete_record_stream = delete_reader.read(Box::pin(delete_stream)).await?;
+
+            while let Some(record_batch) = delete_record_stream.next().await {
+                let record_batch = record_batch?;
+
+                // Position delete files have schema: file_path (string), pos (long)
+                if let Some(file_path_col) = record_batch.column_by_name("file_path")
+                    && let Some(pos_col) = record_batch.column_by_name("pos")
+                {
+                    let file_paths = file_path_col
+                            .as_any()
+                            .downcast_ref::<risingwave_common::array::arrow::arrow_array_iceberg::StringArray>()
+                            .with_context(|| "file_path column is not StringArray")?;
+                    let positions = pos_col
+                            .as_any()
+                            .downcast_ref::<risingwave_common::array::arrow::arrow_array_iceberg::Int64Array>()
+                            .with_context(|| "pos column is not Int64Array")?;
+
+                    // Only include positions that match the current data file
+                    for idx in 0..record_batch.num_rows() {
+                        if !file_paths.is_null(idx) && !positions.is_null(idx) {
+                            let file_path = file_paths.value(idx);
+                            let pos = positions.value(idx);
+
+                            if file_path == data_file_path {
+                                deletes.insert(pos);
+                            }
+                        } else {
+                            tracing::warn!(
+                                "Position delete file {} at line {}: file_path or pos is null",
+                                delete_task_file_path,
+                                idx
+                            );
+                        }
+                    }
+                }
+            }
+        }
+
+        tracing::debug!(
+            "Built position delete set for data file {}: {:?}",
+            data_file_path,
+            deletes
+        );
+
+        deletes
+    } else {
+        BTreeSet::new()
+    };
+
+    // Build equality delete predicates
+    let equality_deletes: Option<(Vec<String>, HashSet<Vec<Datum>>)> = if handle_delete_files {
+        let equality_delete_tasks: Vec<_> = data_file_scan_task
+            .deletes
+            .iter()
+            .filter(|delete| delete.data_file_content == DataContentType::EqualityDeletes)
+            .cloned()
+            .collect();
+
+        if !equality_delete_tasks.is_empty() {
+            // Get equality_ids from the first delete task (should be consistent across all delete files)
+            let equality_ids = equality_delete_tasks[0].equality_ids.clone();
```

Comment:
Actually the equality ids must be the same only holds for iceberg tables produced by RisingWave. External engine can write multiple different equality deletes, so I think we should leave a TODO here to improve it in the future PR.